httpseverywhere.go

```diff
@@ -0,0 +1,262 @@
+package httpseverywhere
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/xml"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/getlantern/golog"
+)
+
+var log = golog.LoggerFor("httpseverywhere")
+
+// ToHTTPS changes an HTTP URL to HTTPS.
+type ToHTTPS interface {
```

Comment:
I ditched the interface (was a bit of a legacy) and went with Rewrite for the method name...